torch/optim/adadelta.py

```diff
@@ -35,6 +35,7 @@ def __init__(self, params, lr=1.0, rho=0.9, eps=1e-6, weight_decay=0):
         defaults = dict(lr=lr, rho=rho, eps=eps, weight_decay=weight_decay)
         super(Adadelta, self).__init__(params, defaults)
 
+    @torch.no_grad()
```

Comment:
This is a clean way of disabling gradient computation within the stepping of the optimizers. It would be nice to be closer to enabling derivatives w/r to parameters of the optimizer. Offline you suggested replacing `@torch.no_grad()` with `with torch.no_grad():` inside the step function. Let's keep it this way for now. It might be easier to do while refactoring anyway, e.g. as part of #30232.

Reply:
We need to have a more high level thinking if we want to enable derivatives wrt the steps.
- What will be the api
- How will the user use this?
- Are the operations in all optimizers actually differentiable (lbfgs for example)

Definitely not in the scope of this PR.

Reply:
Agree.